go/src/koding/kites/kloud/provider/azure/stack.go

```diff
@@ -99,6 +101,14 @@ func (s *Stack) VerifyCredential(c *stack.Credential) error {
 	return nil
 }
 
+func trim(s string, n int) string {
```

Comment:
Creating a separate function just to get a slice of a string? Indeed, there is an extra allocation here. But seriously, this is nothing comparing to other operations/allocations made during stack provisioning.